app/models/product.rb

```diff
@@ -127,7 +127,17 @@ def self.recommended_extensions(root_product_ids)
   end
 
   def service
-    Service.find_or_create_by(product_id: id)
+    service = Service.find_by(product_id: id)
```

Comment:
Is this line needed? Because we have `has_one :service`.

Reply:
Oh, I see :eyes:

Reply:
It used to be `find_or_create_by` -- this line handles the `find` part, rest of the method handles creation of the new service. This is `def service`, so writing `self.service` here would result in recursion -- so I think it is needed.

Reply:
I don't like that `service()` implicitely creates the service if it is not there. We should know at which point we need to create the service and when we can expect it to be there.

Reply:
@digitaltom I don't like `Service` model at all. :-) It doesn't do anything and isn't exposed by the SCC API -- could just use products instead.

There was duplication of service creation code in `RepositoryService` service and `Product` model. Should I move service creation code to `RepositoryService`? Not sure if that will that break, though.

Reply:
@digitaltom OK, I've basically renamed `service()` into `create_service!`, so it's not automagically created on trying to access service anymore but rather explicitly called during SCC sync.